src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -233,78 +204,60 @@ def _format_resource_record_changes(self, action, resource_record):
             'name': record_name,
             'type': record_type
         }
-        response = await self.http_client.get_json(
-            resource_records_url, params=search_params)
-        if response['rrsets']:
-            changes['deletions'] = response['rrsets']
-        return changes
+        return await self.dns_client.get_records_for_zone(
+            zone, params=search_params)
 
-    async def _changes_published(self, change_id, base_changes_url):
+    async def _is_change_done(self, zone, change_id):
```

Comment:
We should put something in the name or docstring to note that there is a timeout applied.